frontend/wasm/src/component/shim_bypass.rs

```diff
@@ -0,0 +1,136 @@
+use midenc_hir::FxHashMap;
+
+use crate::module::module_env::ParsedModule;
+
+/// Checks if a module is a shim module generated by wit-component for indirect lowering.
+///
+/// Shim modules have these characteristics:
+/// 1. Export a table named "$imports"
+/// 2. Export functions (may have numeric names or other patterns)
+/// 3. Functions contain only call_indirect instructions
+/// 4. Usually have minimal imports
+pub fn is_shim_module(module: &ParsedModule) -> bool {
+    log::trace!(
+        "Checking if module is shim module. Exports: {:?}",
+        module.module.exports.keys().collect::<Vec<_>>()
```

Comment:
You can use `midenc_hir::formatter::DisplayValues` here to avoid materializing the `Vec`